packages/vscode-extension/lib/react_devtools_agent.js

```diff
@@ -0,0 +1,39 @@
+let devtoolsAgent = undefined;
+let messageQueue = [];
+
+const agent = {
+  postMessage: (message) => {
+    if (devtoolsAgent) {
+      devtoolsAgent._bridge.send("RNIDE_message", message);
+    } else {
+      messageQueue.push(message);
+    }
+  },
+  onmessage: undefined,
+};
+
+const setDevtoolsAgent = (newDevtoolsAgent) => {
+  if (!newDevtoolsAgent) {
+    return;
+  }
+  devtoolsAgent = newDevtoolsAgent;
+  devtoolsAgent._bridge.addListener("RNIDE_message", (message) => {
+    if (agent.onmessage) {
+      agent.onmessage(message);
+    }
+  });
+  const messages = messageQueue;
+  messageQueue = [];
+  messages.forEach((message) => {
+    devtoolsAgent._bridge.send("RNIDE_message", message);
+  });
+};
+
+const hook = window.__REACT_DEVTOOLS_GLOBAL_HOOK__;
+if (hook.reactDevtoolsAgent) {
+  setDevtoolsAgent(hook.devtoolsAgent);
+} else {
+  hook.on("react-devtools", setDevtoolsAgent);
+}
+
+globalThis.__radon_agent = agent;
```

Comment:
Because it's named `agent`, I initially confused this global to be storing the `reactDevtoolsAgent` when reading the other files. I'd change this to something that's less likely to be confused with anything else.

This class is used as a way for the extension and application to communicate. Maybe something like `RadonExtensionChannel` or `RadonExtensionConnection` would be more fitting?